src/components/Editor.js

```diff
@@ -23,237 +17,118 @@ class Editor extends Component {
     onChange: PropTypes.func.isRequired
   };
 
-  change = (data, name) => {
+  change = (section, prop) => {
     return value => {
-      data[name] = value;
-      this.props.onChange({...this.props.settings});
+      this.props.onChange({
+        ...this.props.settings,
+        [section]: {
+          ...this.props.settings[section],
+          [prop]: value
+        }
+      });
     };
   };
 
+  componentDidUpdate (prevProps) {
+    if (prevProps.settings.__internal.configMode !== this.props.settings.__internal.configMode) {
+      window.componentHandler.upgradeDom();
+    }
+  }
+
   render () {
     const {settings} = this.props;
     const {configMode, platform} = settings.__internal;
     const base = settings.parity.base_path !== '$BASE' ? settings.parity.base_path : basePath(platform);
 
-    const isOffline = settings.parity.mode === 'off';
+    const isOffline = settings.parity.mode === 'offline';
     const isSimple = configMode === 'simple';
 
     return (
       <div>
         { this.select('__internal', 'platform') }
         { this.select('__internal', 'configMode') }
-        <div style={isSimple ? styles.visible : styles.hidden}>
-          { this.renderSimple(settings, platform, base, isOffline) }
-        </div>
-        <div style={!isSimple ? styles.visible : styles.hidden}>
-          { this.renderConfig(settings, platform, base, isOffline) }
+        <div>
```

Comment:
Where did the simplified version go :( IMHO we should auto-generate the full one, but keep the simple one hardcoded.

Reply:
:D 

We still have a simple version (only shows items/sections with `.simple === true` in data.extra.json). The only differences with `master` are the `Parity` & `Footprint` sections coming first and not being collapsible, and the custom `Servers` section on `master` that lists the port/interface settings from the different sections.

- `Parity` & `Footprint` first and not collapsible: we could A) reorder the sections in data.extra.json to make `Parity` and `Footprint` appear first regardless of the configMode, and add `simple_collapsible: false` to those sections in data.extra.json; or B) tweak `renderConfig` to render those sections first and without the collapsible markup.
- custom `Servers` section: we could A) add `simple_section: 'Servers'` to the concerned props in `data.extra.json` and tweak `renderConfig` accordingly (order would be: non-collapsible sections; custom sections; sections and the order would be the same as `master`)

What do you think? We can go another way if you prefer